logprep/abc/processor.py

```diff
@@ -230,8 +232,23 @@ def test_rules(self) -> dict:
 
     @staticmethod
     def resolve_directories(rule_paths: list) -> list:
+        """resolves directories to a list of files or rule definitions
+
+        Parameters
+        ----------
+        rule_paths : list
+            a list of files, directories or rule definitions
+
+        Returns
+        -------
+        list
+            a list of files and rule definitions
+        """
         resolved_paths = []
         for rule_path in rule_paths:
+            if isinstance(rule_path, dict):
+                resolved_paths.append(rule_path)
+                continue
```

Comment:
I have an issue here with the variable naming. The variable `rule_path` does not only contain `paths` anymore, as the list can contain entire rule dictionaries as well now. So it's a mixed data structure. I would have to ideas: (a) we just rename it to something more generic like maybe `rule_descriptions` (as a `file_path`, `uri` or `yaml` could all be seen as a description of a rule) or (b) we actually extend this method, such that the destination for all `file_paths` and `uris` is resolved to the actual rule dictionaries. That way we would only have one list with rule dictionaries. The latter choice would make this method here more complicated, but would simplify the `logprep/processor/base/rule.py::create_rules_from_target` method.